File: app/src/main/java/com/test/kabak/openweather/core/storage/DatabaseManager.java
package com.test.kabak.openweather.core.storage;

import androidx.room.Room;
import android.content.Context;
import androidx.annotation.NonNull;

public class DatabaseManager {
    private static LocalDatabase database;

    public static void init(@NonNull Context context) {
        database = Room
                .databaseBuilder(context, LocalDatabase.class, "database")
                .fallbackToDestructiveMigration()
                .build();
    }

    public static LocalDatabase getDatabase() {
        if (database == null) {
            throw new RuntimeException("Call init() before");
        }

        return database;
    }
}

File: app/src/main/java/com/test/kabak/openweather/core/storage/DatabaseManager.kt
package com.test.kabak.openweather.core.storage

import android.content.Context
import androidx.room.Room
import com.test.kabak.openweather.data.db.LocalDatabase

object DatabaseManager {
    private var database: LocalDatabase? = null

    fun init(context: Context) {
        if (database == null) {
            val applicationContext = context.applicationContext

            database = Room
                    .databaseBuilder(applicationContext, LocalDatabase::class.java, "database")
                    .fallbackToDestructiveMigration()
                    .build()
        }
    }

    fun getDb(): LocalDatabase {
        if (database == null) {
            throw RuntimeException("Call init() first")
        } else {
            return database!!
        }
    }
}